cdap-common/src/main/java/io/cdap/cdap/common/http/AuthenticationChannelHandler.java

```diff
@@ -86,27 +91,21 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
             throw e;
           }
         }
-      } else if (enforceAuthenticatedRequests) {
-        throw new IllegalArgumentException(String.format("Missing Authorization header for request from IP %s",
-                                                         ctx.channel().remoteAddress().toString()));
       }
-
-      SecurityRequestContext.setUserId(currentUserId);
+      LOG.trace("Got user ID '{}' user IP '{}' from IP '{}' and authorization header length '{}'",
+                userID == null ? "NULL" : userID,
+                userIP == null ? "NULL" : userIP,
+                ctx.channel().remoteAddress().toString(),
```

Comment:
No need to call `.toString()`. The slf4j `{}` placeholder will do that when forming the final log message. This is the way to avoid heavy `.toString()` if the given log level is not enabled.

Reply:
Fixed.